### DISCUSSION TITLE
I can't seem to get many to many join query working

### DISCUSSION THREAD

**Discussion by andyjamesn** [supabase/supabase#7651] [Questions] [answered]
I have 2 main tables and a join table

A car can have many features and a feature can have many cars.. hence the many to many.

## cars
id(uuid), name

## features
id(uuid), name

## car_features (Join table)
id(uuid), car_id(fk_cars), feature_id(fk_features)

I am trying to query all cars and get back all cars with all features listed in car_features as follows

```javascript
let { data: styles, error } = await supabase
    .from('cars')
    .select(`
    name,
    features(name)
`)
```

I am basing this on this [documentation](https://supabase.com/docs/reference/javascript/select#query-foreign-tables) but I am getting this error

```shell
code: "PGRST200"
details: null
hint: "Verify that 'cars' and 'features' exist in the schema 'public' and that there is a foreign key relationship between them. If a new relationship was created, try reloading the schema cache."
message: "Could not find a relationship between 'cars' and 'features' in the schema cache"
```

What could I be missing?

**Comment by GaryAustin1** [comment, marked as answer]
Postgrest recently changed and you need to assign each fk of the join table as a primary key (check the primary box on each in the table UI).
https://github.com/supabase/supabase/discussions/7519#discussioncomment-3066861

**Reply by steve-chavez** [reply]
The latest postgREST docs now mention this change: https://postgrest.org/en/latest/api.html#embedding-through-join-tables

I've also made a PR to the Supabase docs to link to latest: https://github.com/supabase/supabase/pull/7656

**Reply by andyjamesn** [reply]
Working perfectly now that I have set the foreign keys as primary keys in my join table. Thanks!

**Comment by jtkeyva** [comment]
Thanks for the info. I'm getting the following error when trying to set my fk as a primary key in my "lead" table:

`Failed to run sql query: constraint "lead_pkey" of relation "lead" does not exist.`

do I need to manually make a column called "lead_pkey"?

this is pretty scary to have a change like this without warning. but that's why it doesn't say production ready correct?

**Reply by steve-chavez** [reply]
> I'm getting the following error when trying to set my fk as a primary key in my "lead" table:

Hm, how are you trying to do that?

> this is pretty scary to have a change like this without warning. 

The frontend/infra teams are working on notifications for breaking changes plus a time window for upgrading services.

**Reply by jtkeyva** [reply]
Thanks for speedy reply. Through the UI by checking the checkbox.

**Reply by jtkeyva** [reply]
i just created a new table and it works but i can't seem to modify my existing table

**Reply by jtkeyva** [reply]
![image](https://user-images.githubusercontent.com/2954865/178634006-31b3eb8f-3e15-43eb-b68a-909c2f0011c9.png)

**Reply by steve-chavez** [reply]
Hm, that could be a bug in the Table UI. Try doing it on the SQL Editor with:

```sql
ALTER TABLE lead 
  DROP CONSTRAINT lead_pkey,
  ADD CONSTRAINT lead_pkey PRIMARY KEY (lead_id, place_id, contact_id);
```

**Reply by chwan97** [reply]
> Hm, that could be a bug in the Table UI. Try doing it on the SQL Editor with:
> 
> ```sql
> ALTER TABLE lead 
>   DROP CONSTRAINT lead_pkey,
>   ADD CONSTRAINT lead_pkey PRIMARY KEY (lead_id, place_id, contact_id);
> ```

big help, to solve this problem I almost remove this error table.

**Comment by Strijdhagen** [comment]
Steve, I believe I found out why this error happens, as I have the same. 
It seems to happen on tables that have the name changed after creation. The change works in the GUI for my tables that did not have the name changed. But for the tabels I have had the name changed I needed to find out the correct names for the _pkey first, drop those, and add them again (all via script)

**Reply by Strijdhagen** [reply]
Similar issue when trying to duplicate the table

**Reply by steve-chavez** [reply]
Good catch @Strijdhagen. That looks like a bug in the UI, could you report it on https://github.com/supabase/supabase/issues?

**Comment by jopfre** [comment]
I also have the same issue when trying to add new primary keys to a renamed table. When the table is renamed through the ui, the pkey name is not updated. I had to delete the old table names pkey with SQL before being able to add new pkeys.
